Odata-docs/webapi-8/fundamentals/entityset-routing.md

```diff
@@ -0,0 +1,545 @@
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Deltas;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using EntitySetRouting.Models;
+
+public class ShapesController : ODataController
+{
+    private static List<Shape> shapes = new List<Shape>
+    {
+        new Rectangle { Id = 1, Length = 7, Width = 4, Area = 28 },
+        new Circle { Id = 2, Radius = 3.5, Area = 38.5 },
+        new Rectangle { Id = 3, Length = 8, Width = 5, Area = 40 }
+    };
+}
+```
+
+## Routing conventions for entity sets
+In this section we cover the conventions for entity set routing and the controller actions (endpoints) required for the requests to be routed successfully.
+
+### Retrieving an entity set
+The route template for this request is: `GET ~/{entityset}`.
+
+The following request returns the `Shapes` entity set - basically a collection of `Shape` entities:
+```http
+GET http://localhost:5000/odata/Shapes
+```
+
+For the above request to be conventionally-routed, a controller action named `Get` (or `GetShapes`) is expected:
+```csharp
+public ActionResult Get()
+{
+    return Ok(shapes);
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Shapes",
+    "value": [
+        {
+            "@odata.type": "#EntitySetRouting.Models.Rectangle",
+            "Id": 1,
+            "Area": 28.0,
+            "Length": 7.0,
+            "Width": 4.0
+        },
+        {
+            "@odata.type": "#EntitySetRouting.Models.Circle",
+            "Id": 2,
+            "Area": 38.5,
+            "Radius": 3.5
+        },
+        {
+            "@odata.type": "#EntitySetRouting.Models.Rectangle",
+            "Id": 3,
+            "Area": 40.0,
+            "Length": 8.0,
+            "Width": 5.0
+        }
+    ]
+}
+```
+
+The response contains 3 shape objects - 2 rectangles and 1 circle. Since `Rectangle` and `Circle` are derived types, each of the shape objects contain an `@odata.type` property specifying the type of the entity.
+
+### Retrieving the count of an entity set
+The route template for this request is: `GET ~/{entityset}/$count`.
+
+To address the raw value of the number of items in an entity set, append `/$count` to that entity set's URL.
+
+The following request returns a count of items in the `Shapes` entity set:
+```http
+GET http://localhost:5000/odata/Shapes/$count
+```
+
+For the above request to be conventionally-routed, a controller action named `Get` (or `GetShapes`) is expected, same as is expected when [retrieving an entity set](#retrieving-an-entity-set). However, the controller action needs to be decorated with `EnableQuery` attribute. The `EnableQuery` attribute is responsible for generating the relevant query for determining the number of items:
```

Comment:
@gathogojr how does the EnableQuery attribute interact with the EnableQueryFeatures configuration?